packages/flutter_tools/test/commands.shard/hermetic/install_test.dart

```diff
@@ -103,16 +103,41 @@ void main() {
       FileSystem: () => fileSystem,
       ProcessManager: () => FakeProcessManager.any(),
     });
+
+    testUsingContext('Passes flavor to application package.', () async {
+      const String flavor = 'free';
+      final InstallCommand command = InstallCommand(verboseHelp: false);
+      command.applicationPackages = FakeApplicationPackageFactory(
+        FakeIOSApp(),
+        getPackageForPlatformInterceptor: (TargetPlatform platform, {File? applicationBinary, BuildInfo? buildInfo}) {
+          expect(buildInfo, isNotNull);
+          expect(buildInfo!.flavor, flavor);
+        },
+      );
+
+      final FakeIOSDevice device = FakeIOSDevice();
+      testDeviceManager.addDevice(device);
+
+      await createTestCommandRunner(command).run(<String>['install', '--flavor', flavor]);
+    }, overrides: <Type, Generator>{
+      Cache: () => Cache.test(processManager: FakeProcessManager.any()),
+      FileSystem: () => fileSystem,
+      ProcessManager: () => FakeProcessManager.any(),
+    });
   });
 }
 
 class FakeApplicationPackageFactory extends Fake implements ApplicationPackageFactory {
-  FakeApplicationPackageFactory(this.app);
+  FakeApplicationPackageFactory(this.app, { this.getPackageForPlatformInterceptor });
 
   final ApplicationPackage app;
+  final void Function(TargetPlatform platform, {BuildInfo? buildInfo, File? applicationBinary})? getPackageForPlatformInterceptor;
```

Comment:
A different way to do this would be to create a writable `BuildInfo` property that gets set in `getPackageForPlatform`, then you can check the `FakeApplicationPackageFactory`'s `buildInfo.flavor` at the end.

Reply:
Yeah, that looks much cleaner 👌